confluent_kafka/src/confluent_kafka.c

```diff
@@ -1187,6 +1187,59 @@ static void error_cb (rd_kafka_t *rk, int err, const char *reason, void *opaque)
 	CallState_resume(cs);
 }
 
+static void throttle_cb (rd_kafka_t *rk, const char *broker_name, int32_t broker_id,
+                         int throttle_time_ms, void *opaque) {
+        Handle *h = opaque;
+        PyObject *ThrottleEvent_type, *throttle_event;
+        PyObject *result, *kwargs, *args;
+        CallState *cs;
+
+        cs = CallState_get(h);
+        if (!h->throttle_cb) {
+                /* No callback defined */
+	        goto done;
+        }
+
+        ThrottleEvent_type = cfl_PyObject_lookup("confluent_kafka",
+                                                 "ThrottleEvent");
+
+        if (!ThrottleEvent_type) {
+                /*ThrottleEvent class not found*/
+                goto err;
+        }
+
+        args = Py_BuildValue("(sid)", broker_name, broker_id, throttle_time_ms);
```

Comment:
You need to cast throttle_time_ms to a double and convert it to seconds:
`(double)throttle_time_ms / 1000.0`

Reply:
Why do we need to convert it to seconds if we are reporting milliseconds

Reply:
got it, python standardizes on seconds